File: demo-androidx/src/main/java/me/yokeyword/sample/demo_flow/ui/fragment/discover/PagerChildFragment.java
package me.yokeyword.sample.demo_flow.ui.fragment.discover;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import me.yokeyword.sample.R;
import me.yokeyword.sample.demo_flow.adapter.PagerAdapter;
import me.yokeyword.sample.demo_flow.base.MySupportFragment;
import me.yokeyword.sample.demo_flow.ui.fragment.CycleFragment;

@SuppressWarnings("FieldCanBeLocal")
public class PagerChildFragment extends MySupportFragment {
    private static final String ARG_FROM = "arg_from";

    private int mFrom;

    private RecyclerView mRecy;
    private PagerAdapter mAdapter;

    public static PagerChildFragment newInstance(int from) {
        final Bundle args = new Bundle();
        final PagerChildFragment fragment = new PagerChildFragment();
        args.putInt(ARG_FROM, from);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        final Bundle args = getArguments();
        if (args != null) {
            mFrom = args.getInt(ARG_FROM);
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.fragment_pager, container, false);
        initView(view);
        return view;
    }

    private void initView(View view) {
        mRecy = view.findViewById(R.id.recy);

        mAdapter = new PagerAdapter(_mActivity);
        final LinearLayoutManager manager = new LinearLayoutManager(_mActivity);
        mRecy.setLayoutManager(manager);
        mRecy.setAdapter(mAdapter);

        mAdapter.setOnItemClickListener((position, view1) -> {
            if (getParentFragment() instanceof DiscoverFragment) {
                ((DiscoverFragment) getParentFragment()).start(CycleFragment.newInstance(1));
            }
        });

        mRecy.post(() -> {
            // Init Datas
            final List<String> items = new ArrayList<>();
            for (int i = 0; i < 20; i++) {
                String item;
                if (mFrom == 0) {
                    item = getString(R.string.recommend) + " " + i;
                } else if (mFrom == 1) {
                    item = getString(R.string.hot) + " " + i;
                } else {
                    item = getString(R.string.favorite) + " " + i;
                }
                items.add(item);
            }
            mAdapter.setDatas(items);
        });
    }
}

File: demo-androidx/src/main/java/me/yokeyword/sample/demo_flow/ui/fragment/discover/PagerChildFragment.kt
package me.yokeyword.sample.demo_flow.ui.fragment.discover

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import me.yokeyword.sample.R
import me.yokeyword.sample.demo_flow.adapter.PagerAdapter
import me.yokeyword.sample.demo_flow.base.MySupportFragment
import me.yokeyword.sample.demo_flow.listener.OnItemClickListener
import me.yokeyword.sample.demo_flow.ui.fragment.CycleFragment
import java.util.*

class PagerChildFragment : MySupportFragment() {
    private var mFrom: Int = 0
    private var mRecy: RecyclerView? = null
    private var mAdapter: PagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val args = arguments
        if (args != null) {
            mFrom = args.getInt(ARG_FROM)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_pager, container, false)
        initView(view)
        return view
    }

    private fun initView(view: View) {
        mRecy = view.findViewById(R.id.recy)
        mAdapter = PagerAdapter(mActivity)
        val manager = LinearLayoutManager(mActivity)
        mRecy!!.layoutManager = manager
        mRecy!!.adapter = mAdapter
        mAdapter!!.setOnItemClickListener(object : OnItemClickListener {
            override fun onItemClick(position: Int, view: View) {
                if (parentFragment is DiscoverFragment) {
                    (parentFragment as DiscoverFragment).start(CycleFragment.newInstance(1))
                }
            }
        })

        mRecy!!.post {
            // Init Datas
            val items = ArrayList<String>()
            for (i in 0..19) {
                val item: String = when (mFrom) {
                    0 -> getString(R.string.recommend) + " " + i
                    1 -> getString(R.string.hot) + " " + i
                    else -> getString(R.string.favorite) + " " + i
                }
                items.add(item)
            }
            mAdapter!!.setDatas(items)
        }
    }

    companion object {
        private const val ARG_FROM = "arg_from"

        fun newInstance(from: Int): PagerChildFragment {
            val args = Bundle()
            val fragment = PagerChildFragment()
            args.putInt(ARG_FROM, from)
            fragment.arguments = args
            return fragment
        }
    }
}
